Library/Homebrew/sandbox.rb

```diff
@@ -63,6 +63,9 @@ def allow_write_temp_and_cache
     allow_write "^/private/var/folders/[^/]+/[^/]+/[C,T]/", :type => :regex
     allow_write_path HOMEBREW_TEMP
     allow_write_path HOMEBREW_CACHE
+    allow_write_path ENV["TMPDIR"] if ENV["TMPDIR"]
+    allow_write_path ENV["TEMP"] if ENV["TEMP"]
+    allow_write_path ENV["TMP"] if ENV["TMP"]
```

Comment:
`HOMEBREW_TEMP` and `HOMEBREW_CACHE` could also be set to a relative path or even random junk if a user wants to. IMO the user should be responsible for making these paths absolute (if set), because it doesn't make sense to have a relative `TMPDIR`.